### PR TITLE
Add ability to search/filter zones

### LINKED ISSUE #602: Quick search/filter zones by Admin Group, Email, Status, etc.
As an administrator with the ability to see a very large quantity of zones, I am looking for the ability to search through zones (in the Zones view) using fields other than the zone name itself.  The ability to filter by Admin Group is desired, in particular.

The quick search box in the top right is clear and effective, and fulfilling this request could possibly involve having that search box to say "keyword" instead of "zone name" -- and then having it search and identify any lines with that keyword in it, period.

An addition suggestion would be having the keyword highlighted in the filtered results.  I would consider that part optional and not truly in scope for this request.

### PR DESCRIPTION
**Pull request by Aravindh-Raju**
Fixes #602. 

Changes in this pull request:
- Added the ability to search/filter zones by admin group.

### CODE REVIEW

**Review by nspadaccino — COMMENTED** [review]
Tested locally and the functionality looks good. Comment about possible enhancement that could be made

File: modules/api/src/main/scala/vinyldns/api/domain/zone/ZoneService.scala [hunk 1]
```diff
@@ -142,33 +143,59 @@ class ZoneService(
...
+        listZonesResult.startFrom,
+        listZonesResult.nextId,
+        listZonesResult.maxItems,
+        listZonesResult.ignoreAccess
+      )
+    }
+    else {
+      for {
+        groupId <- getGroupIdByName(nameFilter.get)
+        listZonesResult <- zoneRepository.listZonesByAdminGroupId(
```
[the remark this answers is not in the document]
Comment (Aravindh-Raju) [reply modules/api/src/main/scala/vinyldns/api/domain/zone/ZoneService.scala]: I have added the wildcard search functionality to this. Now, if we have 3 groups say: test-group, test-teamQA, test-teamDev. We can just search `test` with group name search filter checked and all zones associated with 3 groups will be displayed. We can also use the wildcard character `*est-t*`, and the zones associated with the 2 groups test-teamQA, test-teamDev will be displayed.

**Comment by nspadaccino** [comment]
So I've tested the updates, and while the search functionality seems good when it's a valid search, I'm getting 500 server errors when I search for  an admin group with no results. See screenshot:
Screen Shot 2022-07-14 at 9 52 39 AM
That doesn't happen with an invalid search for zone name, you just get no search results in that case. I'm not sure if this is an issue with my local environment though.

**Comment by Aravindh-Raju** [comment]
> So I've tested the updates, and while the search functionality seems good when it's a valid search, I'm getting 500 server errors when I search for an admin group with no results. See screenshot: Screen Shot 2022-07-14 at 9 52 39 AM That doesn't happen with an invalid search for zone name, you just get no search results in that case. I'm not sure if this is an issue with my local environment though.

Thanks Nick. There was a case in sql query formation which wasn’t handled and resulted in the error. It has been resolved in the latest commit and everything should be working fine now.

**Review by nspadaccino — APPROVED** [review]
Looks good to me!

**Review by Jay07GIT — APPROVED** [review]
LGTM

**Pull request merged**